identity/src/main/java/com/android/identity/mdoc/mso/StaticAuthDataGenerator.java

```diff
@@ -0,0 +1,170 @@
+ *     }
+ *
+ *     DigestIdMapping = {
+ *         NameSpace =&gt; [ + IssuerSignedItemBytes ]
+ *     }
+ *
+ *     ; Defined in ISO 18013-5
+ *     ;
+ *     NameSpace = String
+ *     DataElementIdentifier = String
+ *     DigestID = uint
+ *     IssuerAuth = COSE_Sign1 ; The payload is MobileSecurityObjectBytes
+ *
+ *     IssuerSignedItemBytes = #6.24(bstr .cbor IssuerSignedItem)
+ *
+ *     IssuerSignedItem = {
+ *       "digestID" : uint,                           ; Digest ID for issuer data auth
+ *       "random" : bstr,                             ; Random value for issuer data auth
+ *       "elementIdentifier" : DataElementIdentifier, ; Data element identifier
+ *       "elementValue" : NULL                        ; Placeholder for Data element value
+ *     }
+ * </pre>
+ *
+ * <p>Note that the the byte[] arrays returned in the list of map values are
+ *  the bytes of IssuerSignedItem, not IssuerSignedItemBytes.
+ */
+public class StaticAuthDataGenerator {
+
+    private Map<String, List<byte[]>> mDigestIDMapping = new HashMap<>();
+    private byte[] mEncodedIssuerAuth;
+
+    /**
+     * Constructs a new {@link StaticAuthDataGenerator}.
+     */
+    public StaticAuthDataGenerator() {
+    }
+
+    /**
+     * Sets the <code>DigestIdMapping</code>. This must be called at least once before
+     * generating since <code>DigestIdMapping</code> must be non-empty.
+     *
+     * @param digestIDMapping A non-empty mapping between a <code>Namespace</code> and a list of
+     *                  <code>IssuerSignedItemBytes</code>.
+     * @return The <code>StaticAuthDataGenerator</code>.
+     * @exception IllegalArgumentException if the <code>digestIDMapping</code> is empty.
+     */
+    @NonNull
+    public StaticAuthDataGenerator setDigestIdMapping(@NonNull Map<String, List<byte[]>> digestIDMapping) {
+
+        if (digestIDMapping.isEmpty()) {
+            throw new IllegalArgumentException("digestIDs must not be empty");
+        }
+        mDigestIDMapping = digestIDMapping;
+        return this;
+    }
+
+    /**
+     * Populates the <code>issuerAuth</code> portion of the <code>StaticAuthData</code>. This
+     * contains the bytes of the MobileSecurityObject.
+     *
+     * @param encodedIssuerAuth a COSE_Sign1 object with a payload of MobileSecurityObjectBytes
+     * @return The <code>StaticAuthDataGenerator</code>.
+     */
+    @NonNull
+    public StaticAuthDataGenerator setEncodedIssuerAuth(@NonNull byte[] encodedIssuerAuth) {
+        mEncodedIssuerAuth = encodedIssuerAuth;
+        return this;
+    }
+
+    /**
+     * Builds the <code>StaticAuthData</code> CBOR.
+     *
+     * <p>The resulting byte array can be stored as <code>staticAuthData</code> using
+     * <code>IdentityCredential.storeStaticAuthenticationData({@link X509Certificate},
+     * {@link Calendar}, byte[])</code> and returned using
+     * <code>ResultData.getStaticAuthenticationData()}</code> at presentation time.
```

Comment:
I think we should strip this paragraph as `StaticAuthDataGenerator` will always be generated on the issuer-side, never on the app-side (except for tests and e.g. our example apps with self-signed MDOCs).